samples/test_blinky/src/main.c

```diff
@@ -8,16 +8,15 @@
 #include <zephyr/kernel.h>
 #include <zephyr/drivers/gpio.h>
 
+/* Fail for boards without LED support */
+#if DT_HAS_ALIAS(led0)
+
```

Comment:
Let's put the test next to the error. No reason to split it.

```suggestion
/* Fail for boards without LED support */
#if !DT_HAS_ALIAS(led0)
#error "Compilation halted, board does not have an LED"
#endif
```

Reply:
The reason I did this is because `#error` does not exit the compiler, and thus the rest of the file will spit out a lot of confusing errors due to the missing LED, conflicting with the purpose of a single simple error message. But I can put the error up front by flipping the macro check. Would that be ok?

Reply:
Yup! That'd be fine.